device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/PnpConvention.java

```diff
@@ -121,29 +155,66 @@ public static Message createIotHubMessageUtf8(@NonNull String telemetryName, @No
      *     }
      * }
      */
-    public static Set<Property> createPropertyEmbeddedValuePatch(
+    public static Set<Property> createWritablePropertyResponse(
             @NonNull String propertyName,
             @NonNull final Object propertyValue,
             @NonNull final Integer ackCode,
             @NonNull final Long ackVersion,
             String ackDescription) {
 
-        Map<String, Object> embeddedProperty = new HashMap<String, Object>() {{
-            put("value", propertyValue);
-            put("ac", ackCode);
-            put("av", ackVersion);
-        }};
+        return createWritablePropertyResponse(new HashMap<String, Object>()
+        {{
+            put(propertyName, propertyValue);
+        }}, ackCode, ackVersion, ackDescription);
+    }
 
-        if (ackDescription != null && !ackDescription.isEmpty()) {
-            embeddedProperty.put("ad", ackDescription);
-        }
+    /**
+     * Creates a response to a write request on a device property.
+     * @param propertyPairs The name and unserialized value of the property to report.
+     * @param ackCode The acknowledgment code from the device, for the embedded value property update.
+     * @param ackVersion The version no. of the service-initiated read-write property update.
+     * @param ackDescription (optional) The description from the device, accompanying the embedded value property update.
+     * @return The property patch for embedded value property updates for read-write properties.
+     *
+     * The property patch is created in the below format:
+     * {
+     *     "samplePropertyName": {
+     *         "value": 20,
+     *         "ac": 200,
+     *         "av": 5,
+     *         "ad": "The update was successful."
+     *     }
+     * }
+     */
+    public static Set<Property> createWritablePropertyResponse(
+            @NonNull HashMap<String, Object> propertyPairs,
+            @NonNull final Integer ackCode,
+            @NonNull final Long ackVersion,
+            String ackDescription) {
+
+        Set<Property> response = new HashSet<>();
+
+        propertyPairs.forEach((k,v) ->
+            {
+                if(k == null || k.isEmpty())
+                {
+                    throw new IllegalArgumentException("One of the propertyPairs keys was null, empty, or white space.");
+                }
 
-        return singleton(new Property(propertyName, embeddedProperty));
+                if (ackDescription != null && !ackDescription.isEmpty()) {
+                    response.add(new Property(k, new WritablePropertyResponse(v, ackCode, ackVersion)));
```

Comment:
since ackDescription is not null and is not empty, shouldn't this line include the ackDescription, as such:
response.add(new Property(k, new WritablePropertyResponse(v, ackCode, ackVersion, ackDescription)));

and line 209 without the ackDescription?

Reply:
agreed will change